File: eco-core/core-nms/v1_17_R1/src/main/java/com/willfp/eco/proxy/v1_17_R1/BlockBreak.java
package com.willfp.eco.proxy.v1_17_R1;

import com.willfp.eco.proxy.BlockBreakProxy;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public final class BlockBreak implements BlockBreakProxy {
    @Override
    public void breakBlock(@NotNull final Player player,
                           @NotNull final Block block) {
        player.breakBlock(block);
    }
}

File: eco-core/core-nms/v1_17_R1/src/main/kotlin/com/willfp/eco/proxy/v1_17_R1/BlockBreak.kt
package com.willfp.eco.proxy.v1_17_R1

import org.bukkit.block.Block
import org.bukkit.entity.Player
import proxy.BlockBreakProxy

class BlockBreak : BlockBreakProxy {
    override fun breakBlock(
        player: Player,
        block: Block
    ) {
        player.breakBlock(block)
    }
}